osu.Game/Screens/Play/HUDOverlay.cs

```diff
@@ -381,6 +383,10 @@ public bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
                     }
 
                     return true;
+
+                case GlobalAction.ToggleInGameLeaderboard:
+                    configLeaderboardVisibility.Value = !configLeaderboardVisibility.Value;
```

Comment:
Maybe this reads better? Would also apply to `configSettingsOverlay` if so.

```diff
diff --git a/osu.Game/Screens/Play/HUDOverlay.cs b/osu.Game/Screens/Play/HUDOverlay.cs
index 9c001b3db3..c6f17ec4a1 100644
--- a/osu.Game/Screens/Play/HUDOverlay.cs
+++ b/osu.Game/Screens/Play/HUDOverlay.cs
@@ -78,7 +78,7 @@ protected override bool ShouldBeConsideredForInput(Drawable child)
         public Bindable<bool> ShowHud { get; } = new BindableBool();

         private Bindable<HUDVisibilityMode> configVisibilityMode;
-        private Bindable<bool> configLeaderboardVisibility;
+        private BindableBool configLeaderboardVisibility;
         private Bindable<bool> configSettingsOverlay;

         private readonly BindableBool replayLoaded = new BindableBool();
@@ -180,7 +180,7 @@ private void load(OsuConfigManager config, INotificationOverlay notificationOver
             ModDisplay.Current.Value = mods;

             configVisibilityMode = config.GetBindable<HUDVisibilityMode>(OsuSetting.HUDVisibilityMode);
-            configLeaderboardVisibility = config.GetBindable<bool>(OsuSetting.GameplayLeaderboard);
+            configLeaderboardVisibility = (BindableBool)config.GetBindable<bool>(OsuSetting.GameplayLeaderboard);
             configSettingsOverlay = config.GetBindable<bool>(OsuSetting.ReplaySettingsOverlay);

             if (configVisibilityMode.Value == HUDVisibilityMode.Never && !hasShownNotificationOnce)
@@ -385,7 +385,7 @@ public bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
                     return true;

                 case GlobalAction.ToggleInGameLeaderboard:
-                    configLeaderboardVisibility.Value = !configLeaderboardVisibility.Value;
+                    configLeaderboardVisibility.Toggle();
                     return true;
             }
```

Reply:
Disagree with this one.

Reply:
I do think that reads really well in code, especially in that switch statement. Although what bdach says also makes sense. 

Is casting to a `BindableBool` more common in the code base?

Reply:
It's not a thing in the codebase, no. There's one case of `BindableDouble`, but it's in a test.

https://github.com/ppy/osu/blob/5b44124ecfd4858db94218326495bf8660c72845/osu.Game.Tests/Visual/Gameplay/TestScenePause.cs#L78

I would go safety over a minor readability improvement anyway.